lib/events.js

```diff
@@ -481,23 +468,19 @@ EventEmitter.prototype.removeAllListeners =
     };
 
 EventEmitter.prototype.listeners = function listeners(type) {
-  var evlistener;
-  var ret;
-  var events = this._events;
+  const events = this._events;
 
-  if (!events)
-    ret = [];
+  if (events === undefined)
+    return [];
   else {
-    evlistener = events[type];
-    if (!evlistener)
-      ret = [];
+    const evlistener = events[type];
+    if (evlistener === undefined)
+      return [];
     else if (typeof evlistener === 'function')
-      ret = [evlistener.listener || evlistener];
+      return [evlistener.listener || evlistener];
     else
```

Comment:
The else conditions become obsolete here due to the `return`

Reply:
Speaking of, I think the whole thing can be adjusted to not be an 'else' since the first `if` returns. I'll force push an adjusted commit.

Reply:
That is what I implied :smile: